sdk/eventgrid/Microsoft.Azure.WebJobs.Extensions.EventGrid/src/TriggerBinding/HttpRequestProcessor.cs

```diff
@@ -0,0 +1,78 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Azure.WebJobs.Extensions.EventGrid
+{
+    internal class HttpRequestProcessor
+    {
+        private readonly ILogger _logger;
+
+        public HttpRequestProcessor(ILoggerFactory loggerFactory)
```

Comment:
```suggestion
        public HttpRequestProcessor(ILogger<HttpRequestProcessor> logger)
```

Reply:
We need public `HttpRequestProcessor(ILoggerFactory loggerFactory)` constructor for unit tests

Reply:
you can do `NullLoggerFactory.Instance.CreateLogger<T>()` in unit tests

Reply:
`CreateLogger` does not have generic implementation. It it in some extension?

https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.logging.abstractions.nullloggerfactory.createlogger?view=dotnet-plat-ext-5.0

Reply:
https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.logging.loggerfactoryextensions.createlogger?view=dotnet-plat-ext-5.0